### PR TITLE
cdc: add `RegionId` type to make code more clear

### LINKED ISSUE #17225: cdc introduce `RegionId` type to make the code more clear
At the moment, the region_id type is u64, spread over the codebase, it's hard to maintain and read the code.

So I propose to introduce a new type `RegionId` to wrap up the u64.

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What is changed and how it works?

Issue Number: Close #17225 

What's Changed:

```commit-message
cdc introduce new data type `RegionId` to replace the `u64`, makes the code more easier to read.
```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: components/cdc/src/service.rs [hunk 1]
```diff
@@ -38,6 +38,9 @@ pub struct ConnId(usize);
 #[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
 pub struct RequestId(pub u64);
 
+#[derive(Default, Clone, Copy, Debug, Eq, PartialEq, Hash)]
+pub struct RegionId(pub u64);
```
Comment (overvenus) [inline components/cdc/src/service.rs:42]:
```suggestion
pub struct RegionId(pub u64);
impl Debug for RegionId {
  fn fmt(self, f) -> Result() {
    write!(f, "{}", self.0)
  }
}
```

**Review by overvenus — COMMENTED** [review]
I am hesitant about this PR since other components still use u64, which makes the CDC code style inconsistent with the rest.

**Comment by 3AceShowHand** [comment]
> I am hesitant about this PR since other components still use u64, which makes the CDC code style inconsistent with the rest.

I agree to it, make the region id always `u64` to keep consistent in the code base is much important.

**Comment by 3AceShowHand** [comment]
Close this, since we should keep code style consistency among all components.

If we try to introduce `RegionId`, it should be used among all components.

**Pull request closed without merging**